### PR TITLE
Remove support for backend-specific ops

### PR DESCRIPTION
**Pull request by adk9**
* Remove deprecated backend-specific ops (e.g. CombinedNonMaxSuppression)
* Use opset3 NonMaxSuppression op for NonMaxSuppressionV4
* Remove deprecated backend-specific op for Gather

### CODE REVIEW

File: ngraph_bridge/ngraph_mark_for_clustering.cc [hunk 1]
```diff
@@ -863,6 +862,7 @@ GetTFToNgOpMap() {
         {"Minimum", {std::make_shared<ngraph::opset3::Minimum>()}},
         {"Mul", {std::make_shared<ngraph::opset3::Multiply>()}},
         {"Neg", {std::make_shared<ngraph::op::Negative>()}},
+        {"NonMaxSuppressionV4", {std::make_shared<ngraph::opset3::NonMaxSuppression>(), constant, constant, constant}},
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_mark_for_clustering.cc:865]: I don't think you need to specify constant multiple times in the map.

File: ngraph_bridge/ngraph_builder.cc [hunk 2]
```diff
@@ -2367,25 +2272,10 @@ static Status TranslateGatherV2Op(
...
-    shared_ptr<ng::Node> ng_gather =
-        backend->get_backend_op("Gather", &ng_input, &ng_input_coords, &axis);
-    if (ng_gather == nullptr) {
-      return errors::Internal("In translating GatherV2 op ", op->name(),
-                              " backend could not return valid ngraph node");
-    }
-    Builder::SetTracingInfo(op->name(), ng_gather);
-    SaveNgOp(ng_op_map, op->name(), ng_gather);
-  }
+  auto gather_op = ConstructNgNode<ng::op::Gather>(
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_builder.cc:2275]: Shouldn't this be opset3?

File: ngraph_bridge/ngraph_builder.cc [hunk 3]
```diff
@@ -2945,27 +2835,16 @@ static Status TranslateNonMaxSuppressionV4Op(
...
-  shared_ptr<ng::Node> ng_nmsv4 = backend->get_backend_op(
-      "NonMaxSuppressionV4", &ng_boxes, &ng_scores,
-      (size_t)(max_output_size[0]), (float)(iou_threshold[0]),
-      (float)score_threshold[0], (bool)pad_to_max_output_size);
-  if (ng_nmsv4 == nullptr) {
-    return errors::Internal("In translating NonMaxSuppressionV4 op ",
-                            op->name(),
-                            " backend could not return valid ngraph node");
-  }
+  // ADK: (FIXME) are the outputs set correctly here?
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_builder.cc:2847]: Do we have tests for this op? I couldn't find any, so we should probably add one.

File: ngraph_bridge/ngraph_builder.cc (hunk 2 above)
Comment (adk9) [reply ngraph_bridge/ngraph_builder.cc:2275]: The intent of this PR was to primarily remove backend-specific ops, not upgrade any ops. We'll address upgrade of all Gather ops in a separate PR.

File: ngraph_bridge/ngraph_builder.cc (hunk 3 above)
Comment (adk9) [reply ngraph_bridge/ngraph_builder.cc:2847]: None of the existing backends support this op. And I believe that there are TF unit tests that test this op which we'll enable once we've a backend that supports this op.

Comment (shresthamalik) [reply ngraph_bridge/ngraph_builder.cc:2847]: With this PR we have a translation that is not tested. Better to add support when it could be tested.

Comment (kanvi-nervana) [reply ngraph_bridge/ngraph_builder.cc:2850]: I agree, we should probably only remove the backend specific ops in this PR and add/change support if we can test it.

Comment (adk9) [reply ngraph_bridge/ngraph_builder.cc:2850]: `NonMaxSuppressionV4` is not a backend-specific op. I agree that it's untested at the moment but so are several other ops. I'd rather keep it and fix it as soon as the new backend is up (which is going to be very soon) than comment it out temporarily.

**Pull request merged**